src/ufo.rs

```diff
@@ -66,11 +67,73 @@ impl Default for Ufo {
             groups: None,
             kerning: None,
             features: None,
-            __non_exhaustive: (),
+            data_request: Default::default(),
         }
     }
 }
 
+/// A type that describes which components of a UFO should be loaded.
+///
+/// By default, we load all components of the UFO file; however if you only
+/// need some subset of these, you can pass this struct to [`Ufo::with_fields`]
+/// in order to only load the fields specified in this object.
+///
+/// [`Ufo::with_fields`]: struct.Ufo.html#method.with_fields
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[non_exhaustive]
+pub struct DataRequest {
+    pub layers: bool,
+    pub lib: bool,
+    pub groups: bool,
+    pub kerning: bool,
+    pub features: bool,
+}
+
+impl DataRequest {
+    fn from_bool(b: bool) -> Self {
+        DataRequest { layers: b, lib: b, groups: b, kerning: b, features: b }
+    }
+
+    pub fn all() -> Self {
```

Comment:
Thanks for this, I think it's a nice way of doing the API. It would be nice if these public items had a short doc comment, though?